File: AIM/AIM/Domain/UseCases/UseCaseFactory.m
//
//  UseCaseFactory.m
//  AIM
//
//  Created by Oskar Szydlowski on 04.03.2018.
//  Copyright © 2018 Oskar Szydlowski. All rights reserved.
//

#import "UseCaseFactory.h"

@implementation UseCaseFactory

- (id) initWithOnAirInfoGateway:(id<OnAirInfoGateway>)onAirInfoGateway {
    self = [super init];
    if (self != nil) {
        self.onAirInfoGateway = onAirInfoGateway;
    }
    return self;
}

- (GetOnAirInfoUseCase *)getOnAirInfoUseCase:(OnAirInfoCompletion)completion {
    return [[GetOnAirInfoUseCase new] initWithGateway:self.onAirInfoGateway completion:completion];
}

@end

File: AIM/AIM/Domain/UseCases/UseCaseFactory.swift
//
//  UseCaseFactory.swift
//  AIM
//
//  Created by Oskar Szydlowski on 28.02.2018.
//  Copyright © 2018 Oskar Szydlowski. All rights reserved.
//

protocol UseCaseProducing: class {
    func getOnAirInfoUseCase(completion: @escaping OnAirInfoCompletion) -> GetOnAirInfoUseCase
}

class UseCaseFactory: UseCaseProducing {
    
    let onAirInfoGateway: OnAirInfoGateway
    
    init(onAirInfoGateway: OnAirInfoGateway) {
        self.onAirInfoGateway = onAirInfoGateway
    }
    
    func getOnAirInfoUseCase(completion: @escaping OnAirInfoCompletion) -> GetOnAirInfoUseCase {
        return GetOnAirInfoUseCase(gateway: onAirInfoGateway, completion: completion)
    }
}
